runelite-client/src/main/java/net/runelite/client/plugins/weaponcharges/WeaponChargesPlugin.java

```diff
@@ -0,0 +1,247 @@
+	private static final Pattern ibansCheckString = Pattern.compile("You have ([0-9]+) charges left on the staff.");
+	private static final String ibansRechargeString = "You hold the staff above the well...";
+
+	private WeaponChargesCounter counter;
+
+	private ChargedWeapon rechargedWeapon;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private WeaponChargesOverlay overlay;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	private WeaponChargesConfig config;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Provides
+	WeaponChargesConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(WeaponChargesConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		processChargesCounter();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		removeChargesCounter();
+	}
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return overlay;
+	}
+
+	public Integer getCharges(ChargedWeapon weapon)
+	{
+		if ((weapon == ChargedWeapon.IBANS_STAFF ||
+				weapon == ChargedWeapon.IBANS_STAFF_U) &&
+				config.ibansCharges() >= 0)
+		{
+			return config.ibansCharges();
+		}
+		else
+		{
+			return null;
+		}
+	}
+
+	public Color getChargesColor(int charges)
+	{
+		return charges > 100 ? Color.WHITE : Color.RED;
+	}
+
+	private void addChargesCounter(ChargedWeapon weapon)
+	{
+		if (counter != null)
+		{
+			if (counter.getChargedWeapon() == weapon) return;
+			removeChargesCounter();
+		}
+
+		counter = new WeaponChargesCounter(itemManager.getImage(weapon.getItemId()), this, weapon);
+		infoBoxManager.addInfoBox(counter);
+		log.debug("Added Weapon Charges Counter");
+	}
+
+	private void removeChargesCounter()
+	{
+		if (counter != null)
+		{
+			infoBoxManager.removeInfoBox(counter);
+			counter = null;
+			log.debug("Removed Weapon Charges Counter");
+		}
+	}
+
+	private void processChargesCounter()
+	{
+		final ChargedWeapon equippedWeapon = getEquippedWeapon();
+		if (config.showChargesInfoBox() && equippedWeapon != null)
+		{
+			addChargesCounter(equippedWeapon);
+		}
+		else
+		{
+			removeChargesCounter();
+		}
+	}
+
+	public ChargedWeapon getChargedWeaponFromId(int itemId)
+	{
+		for (ChargedWeapon weapon : ChargedWeapon.values())
+		{
+			if (itemId == weapon.getItemId()) return weapon;
+		}
+
+		return null;
+	}
+
+	private ChargedWeapon getEquippedWeapon()
+	{
+		final ItemContainer equipContainer = client.getItemContainer(InventoryID.EQUIPMENT);
+		if (equipContainer != null)
+		{
+			Item[] equippedItems = equipContainer.getItems();
+			if (equippedItems.length >= WEAPON_SLOT)
+			{
+				return getChargedWeaponFromId(equippedItems[WEAPON_SLOT].getId());
+			}
+		}
+
+		return null;
+	}
+
+	@Subscribe
+	private void onConfigChanged(ConfigChanged event)
+	{
+		processChargesCounter();
+	}
+
+	@Subscribe
+	public void onItemContainerChanged(final ItemContainerChanged event)
+	{
+		processChargesCounter();
```

Comment:
You could probably use the data of the event itself here (and in further methods)